### PR TITLE
fix: logging interface to wrap swift-log

### PR DESCRIPTION
**Pull request by wooj2**
Need to test this a bit more, but i'm pretty sure this will resolve:
https://github.com/awslabs/aws-sdk-swift/issues/518

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by wooj2 — COMMENTED** [review]
Thanks for the review, but this approach, sadly, doesn't work :(

File: Packages/ClientRuntime/Sources/Logging/SwiftLog+LogAgent.swift [hunk 1]
```diff
@@ -6,21 +6,28 @@
...
+    private let logger: Logger
+    private let label: String
+    private var logLevel: Logger.Level
 
-extension SwiftLogger: LogAgent {
-    
-    public var level: LogLevel {
+    public init(label: String) {
+        self.label = label
+        self.logger = Logger(label: label)
```
[the remark this answers is not in the document]
Comment (wooj2) [reply Packages/ClientRuntime/Sources/Logging/SwiftLog+LogAgent.swift:16]: The logger on line 16, `Logger` is of type swift-log

File: Packages/ClientRuntime/Sources/Logging/LogAgent.swift [hunk 2]
```diff
@@ -31,8 +29,17 @@ public protocol LogAgent {
...
+public enum LogAgentLevel: String, Codable, CaseIterable {
+    case trace
+    case debug
+    case info
+    case warn
+    case error
+    case fatal
+}
+
+
```
Comment (kneekey23) [inline Packages/ClientRuntime/Sources/Logging/LogAgent.swift:41]: nit: remove extra new line

File: Packages/ClientRuntime/Sources/Logging/SwiftLog+LogAgent.swift [hunk 3]
```diff
@@ -4,62 +4,66 @@
...
-    
+
     public var name: String {
-            return label
+        return label
     }
     
-    public func log(level: LogLevel,
+    public func log(level: LogAgentLevel,
              message: String,
```
Comment (kneekey23) [inline Packages/ClientRuntime/Sources/Logging/SwiftLog+LogAgent.swift:33]: alignment? format maybe?

**Review by kneekey23 — APPROVED** [review]
LGTM, saw one formatting issue but otherwise ship it!

File: Packages/ClientRuntime/Sources/Logging/SwiftLog+LogAgent.swift (hunk 3 above)
Comment (wooj2) [reply Packages/ClientRuntime/Sources/Logging/SwiftLog+LogAgent.swift:33]: oops. my bad. linting.

**Pull request merged**